pkg/compose/secrets.go

```diff
@@ -30,27 +34,36 @@ import (
 
 func (s *composeService) injectSecrets(ctx context.Context, project *types.Project, service types.ServiceConfig, id string) error {
 	for _, config := range service.Secrets {
-		file := project.Secrets[config.Source]
-		if file.Environment == "" {
-			continue
-		}
-
 		if config.Target == "" {
 			config.Target = "/run/secrets/" + config.Source
 		} else if !isAbsTarget(config.Target) {
 			config.Target = "/run/secrets/" + config.Target
 		}
 
-		env, ok := project.Environment[file.Environment]
-		if !ok {
-			return fmt.Errorf("environment variable %q required by file %q is not set", file.Environment, file.Name)
+		file := project.Secrets[config.Source]
+		var tarArchive bytes.Buffer
+		var err error
+		switch {
```

Comment:
Note: not handling file.Content as this seems to be rejected some other place, causing the printout "validating docker-compose.yml: secrets.content_secret Additional property content is not allowed"

Reply:
I think it comes from the [compose-spec](https://github.com/compose-spec/compose-spec/blob/0d029ec8be0894970d08607972242163224fd28e/schema/compose-spec.json#L733-L782). 
There is no `content` property in `secrets`, only in `configs`. 
This correspond to [the doc](https://docs.docker.com/compose/compose-file/09-secrets/) where secrets can only use `file` or `environment`. 
I assume the reason is to prevent writing secrets value directly in the compose file.